### ISSUE TITLE
Possibility to skip certain api paths or to only handle api subpath

### ISSUE THREAD

**Issue by adsc** [blomqma/next-rest-framework#61]
I have a problem when accessing my swagger doc page. I get two strange errors in the console and no api routes in the swagger doc (probably due to the errors, because when I do the same next-rest-framework setup in an empty next.js project, it works).

The first one is from auth0 callback handler:
```
error - CallbackHandlerError: Callback handler failed. CAUSE: Missing state parameter in Authorization Response.
    at CallbackHandlerError.AuthError [as constructor] (/home/adsc/projects/test1/node_modules/@auth0/nextjs-auth0/dist/utils/errors.js:34:16)
    at CallbackHandlerError.HandlerError [as constructor] (/home/adsc/projects/test1/node_modules/@auth0/nextjs-auth0/dist/utils/errors.js:110:23)
    at new CallbackHandlerError (/home/adsc/projects/test1/node_modules/@auth0/nextjs-auth0/dist/utils/errors.js:138:28)
    at /home/adsc/projects/test1/node_modules/@auth0/nextjs-auth0/dist/handlers/callback.js:49:31
    at step (/home/adsc/projects/test1/node_modules/tslib/tslib.js:195:27)
    at Object.throw (/home/adsc/projects/test1/node_modules/tslib/tslib.js:176:57)
    at rejected (/home/adsc/projects/test1/node_modules/tslib/tslib.js:167:69)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  code: 'ERR_CALLBACK_HANDLER_FAILURE',
  cause: MissingStateParamError: Missing state parameter in Authorization Response.
      at new MissingStateParamError (/home/adsc/projects/test1/node_modules/@auth0/nextjs-auth0/dist/auth0-session/utils/errors.js:26:16)
      at /home/adsc/projects/test1/node_modules/@auth0/nextjs-auth0/dist/auth0-session/handlers/callback.js:24:63
      at step (/home/adsc/projects/test1/node_modules/tslib/tslib.js:195:27)
      at Object.next (/home/adsc/projects/test1/node_modules/tslib/tslib.js:176:57)
      at fulfilled (/home/adsc/projects/test1/node_modules/tslib/tslib.js:166:62)
      at processTicksAndRejections (node:internal/process/task_queues:96:5) {
    expose: true,
    statusCode: 404,
    status: 404
  },
  status: 404,
  page: '/api/auth/callback'
}
```

The second error is pretty weird as well. It's a dynamic api route allowing to pass numeric id as last url segment, and it works fine as an api, but somehow when next-rest-framework is handling it, this error is produced:
```
error - PrismaClientValidationError: 
Invalid `prisma.sample.findUnique()` invocation:

{
  where: {
    id: NaN
        ~~~
  }
}

Argument id: Got invalid value NaN on prisma.findUniqueSample. Provided Float, expected Int.


    at Ui.validate (/home/adsc/projects/test1/node_modules/@prisma/client/runtime/library.js:148:73)
    at gn.createMessage (/home/adsc/projects/test1/node_modules/@prisma/client/runtime/library.js:164:1298)
    at /home/adsc/projects/test1/node_modules/@prisma/client/runtime/library.js:177:11600
    at Object.runInChildSpan (/home/adsc/projects/test1/node_modules/@prisma/client/runtime/library.js:174:1394)
    at t._executeRequest (/home/adsc/projects/test1/node_modules/@prisma/client/runtime/library.js:177:11577)
    at async handler (webpack-internal:///(api)/./pages/api/genetic/sample/[id].ts:17:22)
    at async Object.apiResolver (/home/adsc/projects/test1/node_modules/next/dist/server/api-utils/node.js:372:9)
    at async DevServer.runApi (/home/adsc/projects/test1/node_modules/next/dist/server/next-server.js:488:9)
    at async Object.fn (/home/adsc/projects/test1/node_modules/next/dist/server/next-server.js:751:37)
    at async Router.execute (/home/adsc/projects/test1/node_modules/next/dist/server/router.js:253:36)
    at async DevServer.run (/home/adsc/projects/test1/node_modules/next/dist/server/base-server.js:384:29)
    at async DevServer.run (/home/adsc/projects/test1/node_modules/next/dist/server/dev/next-dev-server.js:743:20)
    at async DevServer.handleRequest (/home/adsc/projects/test1/node_modules/next/dist/server/base-server.js:322:20) {
  clientVersion: '4.16.2',
  page: '/api/genetic/sample/[id]'
}
```
I added a console.log to see what the actual value is for the id parameter, and it's "{id}". When I call the api with url api/genetic/sample/1, I get a correct "1" there instead.

So it would be helpful for me, if there was a possibility to exclude certain api paths from being handled by next-rest-framework, or maybe to only include those from a specified path.

**Comment by adsc** [comment]
I was able to solve the latter problem by adding more checks to the extraction of the id request parameter.

**Comment by adsc** [comment]
Hmmm, it popped up again. Here is the code of an example api route.

pages/api/media/[id].ts:
```typescript
import { PrismaClient } from "@prisma/client";
import { NextApiRequest, NextApiResponse } from "next";

const prisma = new PrismaClient()

export default async function handler(
   req: NextApiRequest,
   res: NextApiResponse
) {
   const { id } = req.query

   if (!id || Array.isArray(id) || Number.isNaN(id)) {
      res.status(400).statusMessage = "No id present in request query or multiple ids given"
      return res.end()
   }

   const media = await prisma.media.findUnique({
      where: {
         id: Number(id),
      },
   })

   if (!media) {
      return res.status(404).end()
   }

   return res.status(200).json(media)
}
```

Here is my next-rest-framework client.ts:
```typescript
import { NextRestFramework } from 'next-rest-framework';

export const { defineCatchAllApiRoute, defineApiRoute } = NextRestFramework({
   apiRoutesPath: 'pages/api',
   openApiSpecOverrides: {
      components: {
         securitySchemes: {
            bearerAuth: {
               type: "http",
               scheme: "bearer",
            },
         },
      },
      security: {
         bearerAuth: [],
      },
   },
});
```

**Comment by blomqma** [comment]
> Hmmm, it popped up again. Here is the code of an example api route.
> 
> pages/api/media/[id].ts:
> 
> ```ts
> import { PrismaClient } from "@prisma/client";
> import { NextApiRequest, NextApiResponse } from "next";
> 
> const prisma = new PrismaClient()
> 
> export default async function handler(
>    req: NextApiRequest,
>    res: NextApiResponse
> ) {
>    const { id } = req.query
> 
>    if (!id || Array.isArray(id) || Number.isNaN(id)) {
>       res.status(400).statusMessage = "No id present in request query or multiple ids given"
>       return res.end()
>    }
> 
>    const media = await prisma.media.findUnique({
>       where: {
>          id: Number(id),
>       },
>    })
> 
>    if (!media) {
>       return res.status(404).end()
>    }
> 
>    return res.status(200).json(media)
> }
> ```
> 
> Here is my next-rest-framework client.ts:
> 
> ```ts
> import { NextRestFramework } from 'next-rest-framework';
> 
> export const { defineCatchAllApiRoute, defineApiRoute } = NextRestFramework({
>    apiRoutesPath: 'pages/api',
>    openApiSpecOverrides: {
>       components: {
>          securitySchemes: {
>             bearerAuth: {
>                type: "http",
>                scheme: "bearer",
>             },
>          },
>       },
>       security: {
>          bearerAuth: [],
>       },
>    },
> });
> ```

Okay, so your second issue seems to originate from the same reason, with Next REST Framework calling your non-NRF route when generating the OpenAPI spec, resulting in the error you described. I have now added support for setting allow/deny rules for routes, allowing you to deny the `pages/api/media/[id].ts` path from Next REST Framework. This fix is released with the latest [v1.2.0](https://github.com/blomqma/next-rest-framework/releases/tag/v1.2.0) release.
